File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {boolean}
 */
var canMakeSubsequence = function (str1, str2) {
    let alpha = 'abcdefghijklmnopqrstuvwxyza';
    let i = 0, j = 0;
    let n = str1.length, m = str2.length;

    while (i < n && j < m) {
        if (str1[i] === str2[j]) {
            j++;
        }
        else if (String.fromCharCode((str1[i].charCodeAt(0) - 'a'.charCodeAt(0) + 1) % 26 + 'a'.charCodeAt(0)) === str2[j]) {
            j++;
        }
        i++;
    }

    return j == m;
};

File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        alpha = 'abcdefghijklmnopqrstuvwxyza'
        i, j = 0, 0
        n, m = len(str1), len(str2)

        while i < n and j < m:
            if str1[i] == str2[j]:
                j += 1
            elif alpha[(alpha.index(str1[i]) + 1) % 26] == str2[j]:
                j += 1
            i += 1

        return j == m